src/track/beatfactory.cpp

```diff
@@ -84,9 +84,22 @@ mixxx::BeatsPointer BeatFactory::makePreferredBeats(
     const QString version = getPreferredVersion(fixedTempo);
     const QString subVersion = getPreferredSubVersion(extraVersionInfo);
 
+#ifdef DEBUG_PRINT_BEATS
```

Comment:
I like to keep it like that because currently the code is grayed out by the IDE similar as it is if it is commented.
This way it does not drew attraction when reading the code.

I don't mind that it may fail if I activate it later, because I take this more as a template for future debug attempts, where I might looking into a different topic and the statements will look differently. 
I like to keep this in this case, because I have already written three times similar code here and this will hopefully save me from some work.

I know you did s more sophisticated approach in the library region, but to be honest I have never used the debug switches there, because I am not aware of all implications. 
Instead I always comment out the guards to have my debug statement fully under control.